### PR TITLE
adding clean_up methods to basic and unique_id tests

### LINKED ISSUE #9219: [CT-3459] [Feature] Add cleanup methods to various base tests in core that are known to create leftover schemas from ci/cd in adapters
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

Going to clear out our test account used for dbt-redshift see that several tests defined in dbt-core are leaving hanging artifacts at a rate suggesting their not properly being dropped see: https://github.com/dbt-labs/dbt-redshift/issues/675

### Describe alternatives you've considered

We can go look at the tests and evaluate what part of the default schema Is being changed and create cleanup methods to drop that changed schema. then install the connecting pr branch to this issue into local adapter runs and run tests locally to make sure the caught tests are not leaving any more schemas and try to catch follow-up ones that may not be nuisance artifacts

### Who will this benefit?

our testing alerts can hopefully start to be more clean and pointing to real failures and stem flow of slowly filling up our ci/cd databases till no tests are allowed to pass

### Are you interested in contributing this feature?

yes

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by McKnight-42**
resolves #9219, dbt-redshift- https://github.com/dbt-labs/dbt-redshift/issues/675

### Problem
found in dbt-redshift but possibly extending to all adapters some hanging artifacts found in tests imported from core within the `test_basic` and `incremental_unique_id` files.

### Solution
testing if adding `cleanup` methods to try and guarantee dropping of schemas post test runs

### Testing Process
- after adding the cleanup methods I went and clear out redshift in postico and ran the tests locally several times in multi and single thread on Thursday/Friday of last week to test for other hanging artifacts and see make sure they were catching correct allocations. 

- on Monday 12/3 I again checked postico on redshift to get idea of level of build up over a few days and cleared and re tested, then went and cleared out bigquery as well and tested there.

after conversation around `docs_genreate` test failure I am again testing adapters
dbt-redshift seems to behave as expected in local multi-thread runs ✅ 
dbt-redshift local single thread ✅ 
ci/cd test draft pr: https://github.com/dbt-labs/dbt-redshift/pull/683 (ran 3.8 integration test twice and both times ended no hanging artifacts) ✅ 

dbt-bigquery local mulit-thread ✅ 
dbt-bigquery local single thread ✅ 
dbt-bigquery draft pr for ci/.cd run https://github.com/dbt-labs/dbt-bigquery/pull/1049 ✅ 

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [ ] I have run this code in development and it appears to resolve the stated issue  
- [ ] This PR includes tests, or tests are not required/relevant for this PR
- [ ] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [ ] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

**Comment by McKnight-42** [comment]
PR discussion:
- Is there any further level of testing we want to do before approval?
- Is this something we want to become a catch-all for other possible hanging tests in other databases?

**Review by colin-rogers-dbt — APPROVED** [review]

File: tests/adapter/dbt/tests/adapter/basic/test_docs_generate.py
Comment (colin-rogers-dbt) [inline tests/adapter/dbt/tests/adapter/basic/test_docs_generate.py:407]: Looks like this might have broken the docs tests? https://github.com/dbt-labs/dbt-core/actions/runs/7065010846/job/19307239764?pr=9195#step:8:432

Could be an inter-test dependency (i.e. docs generate was relying on some artifacts _not_ being cleaned up)

Comment (McKnight-42) [reply tests/adapter/dbt/tests/adapter/basic/test_docs_generate.py:407]: going to try and swap over to specifying `unqiue_schema` instead of `test_schema` in clean_up method and see if that fixes it `test_schema` should ideally be being picked up by the default teardown method we have designed anyway.

**Review by colin-rogers-dbt — APPROVED** [review]

File: tests/functional/artifacts/data/state/v12/manifest.json
Comment (emmyoop) [inline tests/functional/artifacts/data/state/v12/manifest.json:1]: Why do these changes require manifest changes?

File: .changes/unreleased/Under the Hood-20231211-112858.yaml
Comment (emmyoop) [inline .changes/unreleased/Under the Hood-20231211-112858.yaml:1]: Changes to just tests should not have a changelog.

**Review by emmyoop — CHANGES_REQUESTED** [review]
Generally looks good.  I have comments on 2 files.

Can you also add some notes to https://github.com/dbt-labs/dbt-core/issues/9111 so we track this on a larger scale?

File: tests/functional/artifacts/data/state/v12/manifest.json
Comment (McKnight-42) [reply tests/functional/artifacts/data/state/v12/manifest.json:1]: honestly, this seemed to change every time I did a git add . i didn't manually change anything in it

Comment (emmyoop) [reply tests/functional/artifacts/data/state/v12/manifest.json:1]: Oh no!  That's because [this line](https://github.com/dbt-labs/dbt-core/blob/6e0a38720529360d061bd4e24e140f5ec55f8c15/tests/functional/artifacts/test_previous_version_state.py#L415) is uncommented!  It should be commented out.  Looks like it got accidentally merged.

Comment (emmyoop) [reply tests/functional/artifacts/data/state/v12/manifest.json:1]: Can you comment it out and delete your manifest generated?

Comment (McKnight-42) [reply tests/functional/artifacts/data/state/v12/manifest.json:1]: will do

Comment (McKnight-42) [reply tests/functional/artifacts/data/state/v12/manifest.json:1]: okay after deleting the line, and the manifest we would have some failures in the test for not having a manifest to compare to so I grabbed the one generated from the above pr.

Comment (emmyoop) [reply tests/functional/artifacts/data/state/v12/manifest.json:1]: Sorry, I meant to revert the changes.

Comment (emmyoop) [reply tests/functional/artifacts/data/state/v12/manifest.json:1]: Can you revert the file instead of generating a new one?

Comment (McKnight-42) [reply tests/functional/artifacts/data/state/v12/manifest.json:1]: I can try i just copy/pasted it from Quigley's pr

**Review by emmyoop — APPROVED** [review]
Look good.

**Pull request merged**
